medium-tests/src/test/java/mediumtest/TelemetryMediumTests.java

```diff
@@ -179,7 +178,7 @@ void it_should_ping_telemetry_endpoint() throws ExecutionException, InterruptedE
         "  \"ide_version\" : \"4.5.6\",\n" +
         "  \"platform\" : \"linux\",\n" +
         "  \"architecture\" : \"x64\",\n" +
-        "  \"connected_mode_used\" : true,\n" +
+        "  \"connected_mode_used\" : false,\n" +
```

Comment:
Why is it not `true` ? There is a connection and a binding declared.

Reply:
This is an interesting one.  We set a `System.setProperty("sonarlint.internal.telemetry.initialDelay", "0")` at the start of the test so we don't need to wait for the telemetry scheduling to start in one minute (that's the default).
In this case, this means the telemetry endpoint is contacted pretty much immediately at the point when we build the backend - even before the [configuration is added](https://github.com/SonarSource/sonarlint-core/blob/task/vs/slcore-628_get_telemetry_from_core/medium-tests/src/test/java/mediumtest/fixtures/SonarLintBackendFixture.java#L353).

So the test used to return true just because that's what we mocked the client to return in the live attributes 
` when(fakeClient.getTelemetryLiveAttributes()).thenReturn(new TelemetryLiveAttributesResponse(true, false, null, false, emptyList(), emptyList(), emptyMap()));`  (and declaring the connection and binding on the backend didn't have impact).
Since the `connected_mode_used` value is now computed on the backend, it got correctly computed to false as there is no binding at the moment of the computation.
Rather than trying to fiddle with the initial scheduling delay for Telemetry I think it's ok to have `false` returned for this test (as the actual computation is tested by other tests). I will just remove the `.....withSonarQubeConnection("connectionId").withBoundConfigScope("scopeId", "connectionId", "projectKey")` bit so it's not confusing.

Reply:
Sounds good to me, this will be less confusing indeed!